include/CL/sycl/parallelism/detail/parallelism.hpp

```diff
@@ -283,7 +283,7 @@ void parallel_for_workgroup(nd_range<Dimensions> r,
 template <std::size_t Dimensions, typename ParallelForFunctor>
 void parallel_for_workitem(const group<Dimensions> &g,
                            ParallelForFunctor f) {
-#if defined(_OPENMP) && !defined(TRISYCL_NO_BARRIER)
+#if defined(_OPENMP) && (!defined(TRISYCL_NO_BARRIER) && !defined(_MSC_VER))
```

Comment:
Barriers will never be fast on CPU.
This is why we are banning them when possible with the use of hierarchical parallelism.
That said, I have the feeling this code could be rewritten with `#pragma omp for` instead.
Otherwise, you can always compile SYCL without OpenMP support in the meantime...
Yes you are right, we may run in an explosion complexity if we have to target many standards